### PR TITLE
Updated entirety of fixnum package to adhere to Effective Dart style standards

### PR DESCRIPTION
**Pull request by somarkoe**
Much of this package doesn't follow Effective Dart's style guide, so I updated everything such that it could be more readable!

Rationale:
* [Updating const names to lowerCamelCase](https://dart.dev/guides/language/effective-dart/style#prefer-using-lowercamelcase-for-constant-names)
  * Old values were deprecated, there are no breaking changes from this PR. 
* [Un-abbreviating variable names](http://go/effective-dart/design#avoid-abbreviations)
* [Keeping low, middle, and high at the end of variable names](http://go/effective-dart/design#prefer-putting-the-most-descriptive-noun-last)
  * I figure the location of the chunk is the "most descriptive" piece as well as the most important  part, but can also swap this around, if needed; struggled a bit with this one!
* [Updating variables from "a", "b", "x", etc.](http://go/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable) 
* I also slightly adjusted the "return type" logic in division, since using const numbers like an enum is brittle. Updated this to use an actual enum such that it is backed by the analyzer.

### CODE REVIEW

**Comment by kevmoo** [comment]
Did you look at adding/updating analysis_options.yaml here?

**Comment by kevmoo** [comment]
Enabling `constant_identifier_names` might be good

**Comment by somarkoe** [comment]
Done!

Disabled the lint for the deprecated members.

**Comment by samlythemanly** [comment]
@kevmoo LMK if there's anything else that needs to be done here 😁

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 1.0.1
+
+* Updated code to adhere to
```
Comment (kevmoo) [inline CHANGELOG.md:3]: This would be a `1.1.0` release, since we have new APIs.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
...
+[Effective Dart](https://dart.dev/guides/language/effective-dart).
```
Comment (kevmoo) [inline CHANGELOG.md:4]: Be clear what you're changing. Created a bunch of new members and deprecated old ones, etc.

**Review by kevmoo — CHANGES_REQUESTED** [review]

**Review by rakudrama — COMMENTED** [review]
I am not totally on-board with these changes but it will take me some days to review.

File: lib/src/int64.dart [hunk 3]
```diff
@@ -199,264 +229,280 @@ class Int64 implements IntX {
...
+    int cHighPartialLow = partialMiddle >> 18;
+    int cHighPartialMiddle = partialHigh >> 5;
+    int cHighPartialHigh = (partialHighest & 0xfff) << 8;
+    int cHigh = cHighPartialLow + cHighPartialMiddle + cHighPartialHigh;
 
     // Propagate high bits from c0 -> c1, c1 -> c2.
-    c1 += c0 >> _BITS;
-    c2 += c1 >> _BITS;
+    cMiddle += cLow >> _lowBitCount;
+    cHigh += cMiddle >> _lowBitCount;
```
Comment (rakudrama) [inline lib/src/int64.dart:331]: If we are to name the bit counts by position, then we need a separate name for the two different uses of `_BITS`, e.g. `_lowBitCount` and `_middleBitCount`. Otherwise the use of `_low` here looks like a mistake.

File: lib/src/int64.dart [hunk 4]
```diff
@@ -10,127 +10,157 @@
...
+  static const int _highBitCount = 64 - _lowAndMiddleBitCount; // 20
+  static const int _mask = (1 << _lowBitCount) - 1; // 4194303
+  static const int _maskHigh = (1 << _highBitCount) - 1; // 1048575
+  static const int _signBit = _highBitCount - 1; // 19
+  static const int _signBitMask = 1 << _signBit;
 
   /// The maximum positive value attainable by an [Int64], namely
   /// 9,223,372,036,854,775,807.
-  static const Int64 MAX_VALUE = Int64._bits(_MASK, _MASK, _MASK2 >> 1);
+  @Deprecated('Use [minValue] instead.')
```
Comment (rakudrama) [inline lib/src/int64.dart:41]: Use [maxValue] instead.

File: lib/src/int64.dart [hunk 5]
```diff
@@ -869,31 +918,41 @@ class Int64 implements IntX {
...
-    int diff2 = a2 - b2 - ((diff1 >> _BITS) & 1);
-    return _masked(diff0, diff1, diff2);
+  static Int64 _masked(int low, int middle, int high) =>
+      Int64._bits(_mask & low, _mask & middle, _maskHigh & high);
+
+  static Int64 _subtract(
+      int firstMinuendLow,
+      int firstMinuendMiddle,
+      int firstMinuendHigh,
+      int secondMinuendLow,
```
Comment (rakudrama) [inline lib/src/int64.dart:931]: This would be the subtrahend, not minuend, but this change is too much churn.

File: lib/src/int64.dart [hunk 6]
```diff
@@ -199,264 +229,280 @@ class Int64 implements IntX {
...
 
-    return Int64._masked(res0, res1, res2);
+    return Int64._masked(resultLow, resultMiddle, resultHigh);
   }
 
   @override
-  Int64 shiftRightUnsigned(int n) {
-    if (n < 0) {
-      throw ArgumentError.value(n);
+  Int64 shiftRightUnsigned(int number) {
```
Comment (rakudrama) [inline lib/src/int64.dart:458]: `number` -> `count`

File: lib/src/int64.dart [hunk 7]
```diff
@@ -199,264 +229,280 @@ class Int64 implements IntX {
...
 
-    return Int64._masked(res0, res1, res2);
+    return Int64._masked(resultLow, resultMiddle, resultHigh);
   }
 
   @override
-  Int64 operator >>(int n) {
-    if (n < 0) {
-      throw ArgumentError.value(n);
+  Int64 operator >>(int number) {
```
Comment (rakudrama) [inline lib/src/int64.dart:409]: `number` -> `count`

**Review by rakudrama — CHANGES_REQUESTED** [review]
I like some of the _public_ changes, like providing the constant `zero` and some of the parameter name changes in public methods.

A lot of important customers rely on this package, so we want to minimize churn. We need to be extremely cautious about changes in the more 'mathematical' parts of the code.

Legacy elements still need to be tested. Adding `zero` does not mean that tests using `ZERO` can be changed - we need to ensure that customers relying on `ZERO` still have the same level  of testing, even if they theoretically should migrate to `zero`.

We can't mark `ZERO` as deprecated, since it is used in generated code (for protocol buffers), and the generated code does not suppress that lint. We would need to fix the protocol buffer code generator first. Deprecating ZERO is a multi-step process that will take months. I don't think it is worth it right now.

I would be happy to review a version of this change that is restricted to the public API and adds tests rather than renames them, but this change is just too much code churn as it stands.

File: lib/src/int64.dart [hunk 8]
```diff
@@ -199,264 +229,280 @@ class Int64 implements IntX {
...
-    if (b1 != 0) {
-      p1 += a0 * b1;
-      p2 += a1 * b1;
-      p3 += a2 * b1;
-      p4 += a3 * b1;
+    int partialLowest = lowest * operandLowest; // << 0
+    int partialLow = low * operandLowest; // << 13
+    int partialMiddle = middle * operandLowest; // << 26
+    int partialHigh = high * operandLowest; // << 39
+    int partialHighest = highest * operandLowest; // << 52
```
[the remark this answers is not in the document]
Comment (samlythemanly) [reply lib/src/int64.dart:274]: Yeah it was difficult trying to come up with names that followed that line item, but also didn't stretch too far into longNamesAreLong-land. However, I'm not entirely opposed to `middlePartialProduct`. 

As far as the symmetry between a & b, totally agree there.

However, when the code was littered with single letter variables, as someone who has had to step through the code, it was near impossible to discern what it was doing.

[the remark this answers is not in the document]
Comment (samlythemanly) [reply lib/src/int64.dart:274]: Regarding churn, I can definitely break this into more palatable chunks (perhaps closer to a method-by-method basis) as well! Just let me know :)

[the remark this answers is not in the document]
Comment (rakudrama) [reply lib/src/int64.dart:274]: Let's keep the changes restricted to things which benefit developers using the package.

1. New aliases for constants which conform to the guidelines, i.e. add `zero` and friends.
2. Improving the names of parameters of methods. This is beneficial in code completion.
3. Tests which ensure both `ZERO` and `zero` are equivalent (and likewise for other aliases).

These changes require bumping the version as described by kevmoo.
